### DISCUSSION TITLE
ScaledJob "Target Average Value" prometheus

### DISCUSSION THREAD

**Discussion by rh-voice** [kedacore/keda#3437] [General]
Hi there,

So I am trying to scale some jobs that run agents for our CI/CD; I can read pipeline demand metrics from prometheus (I have an exporter for that); Now this works fine with ScaledObjects; but I am trying to switch those to jobs.

My question is; what the "Target Average Value" attribute for prometheus ? because Threshold is not working; which is expected; it defines the threshold to start scaling for HPA not for spinning up jobs.

**Comment by JorTurFer** [comment]
Hey,
Sorry, I don't get you question, could you redo it?. The threshold is the target value per instance that the autoscaling tries to reach, if you set ie `threshold: 5` and your prometheus returns 20, hypothetically you should have 4 instances, it's not something for starting from it. If you need that, we are introducing a new activation feature for next release, but current threshold is the amount of X per instance in jobs

**Reply by rh-voice** [reply]
Hmm, ok that makes sense. I am guessing its something in my polling interval; it might be polling faster than than the cooling down of that value.

could you elaborate on that new activation feature ? I've always wanted something webhook-like for scaledJobs; just an endpoint for the scaledJob to hit and request a job; that actually would solve my issue

**Reply by JorTurFer** [reply]
No no, the activation feature is not for that, it's just to not scaling from/to zero only if queue length is > 0, basically, because maybe you are not interested on having 1 instance only for 1 message and you could want to not scale from zero even you have any pending message.
This is [the issue](https://github.com/kedacore/keda/issues/2800) and this is [the doc](https://keda.sh/docs/2.8/concepts/scaling-deployments/#activating-and-scaling-thresholds).
BTW, if you want to have and endpoint you can call to spawn jobs, you could use Metrics API and create the logic in an external REST API that KEDA calls using [Metrics API scaler](https://keda.sh/docs/2.7/scalers/metrics-api/)
